superset/models/core.py

```diff
@@ -113,6 +114,32 @@ class ConfigurationMethod(StrEnum):
     DYNAMIC_FORM = "dynamic_form"
 
 
+class EngineManager:
+    """
+    Allocate an engine once per process as per API documentation
+    https://docs.sqlalchemy.org/en/20/core/connections.html
+
+    Quote:
+      The typical usage of create_engine() is once per particular database URL,
+      held globally for the lifetime of a single application process
+    """
+
+    _lock = threading.Lock()
+    _sqla_engines: dict[str, tuple[Engine, dict]] = {}
+
+    @classmethod
+    def create_engine(cls, sqlalchemy_url: str, **params) -> Engine:
+        with cls._lock:
+            tpl = cls._sqla_engines.get(sqlalchemy_url)
+            if tpl is not None:
+                engine, cparams = tpl
+                if cparams == params:
```

Comment:
Can we have `params` be part of the key as well, in some deterministic serialized form? A few databases that support user impersonation will add the user info to `params`, and in that case the engine manager would not be very useful.

Reply:
PTAL